### DISCUSSION TITLE
blog plugin build error :Expected metadata to be defined but found nothing

### DISCUSSION THREAD

**Discussion by renyuanzhe** [squidfunk/mkdocs-material#6499] [Q&A] [answered]
Is it necessary that each .md document has all the metadata required?

(blog) D:\Users\Claymore\Desktop\blog\my_blog>mkdocs build
INFO    -  Cleaning site directory
INFO    -  Building documentation to directory: D:\Users\Claymore\Desktop\blog\my_blog\site
ERROR   -  Error reading metadata of post 'blog\posts\我的笔记 d29b9c0a044341cfa70d660a49aff04c.md' in 'docs':
           Expected metadata to be defined but found nothing

Aborted with a BuildError!

**Comment by alexvoss** [comment, marked as answer]
AFAIK the only metadata you need is the date, which is a pretty minimal requirement. Note that it is good practice to define metadata such as the publication date in the file rather than taking it from the filesystem. If you have additional metadata and don't want to define it for each post then you can [use the meta plugin from the Insiders edition](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/?h=blog#setting-defaults).

**Reply by renyuanzhe** [reply]
thankyou
